quickwit-actors/src/mailbox.rs

```diff
@@ -0,0 +1,271 @@
+    type Target = Inner<Message>;
+
+    fn deref(&self) -> &Self::Target {
+        self.inner.as_ref()
+    }
+}
+
+pub struct Inner<Message> {
+    pub(crate) sender: flume::Sender<ActorMessage<Message>>,
+    command_sender: flume::Sender<Command<Message>>,
+    id: Uuid,
+    actor_name: String,
+}
+
+pub enum Command<Msg> {
+    Pause,
+    Stop(oneshot::Sender<()>),
+    Start,
+    Observe(oneshot::Sender<()>),
+    //< scheduled message go through the command channel as they are prioritory.
+    ScheduledMessage(Msg),
+}
+
+impl<Msg> fmt::Debug for Command<Msg> {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        match self {
+            Command::Pause => write!(f, "Pause"),
+            Command::Stop(_) => write!(f, "Stop"),
+            Command::Start => write!(f, "Start"),
+            Command::Observe(_) => write!(f, "Observe"),
+            Command::ScheduledMessage(_) => write!(f, "ScheduleMsg"),
+        }
+    }
+}
+
+impl<Message: fmt::Debug> fmt::Debug for Mailbox<Message> {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        write!(f, "Mailbox({})", self.actor_instance_name())
+    }
+}
+
+impl<Message> Hash for Mailbox<Message> {
+    fn hash<H: std::hash::Hasher>(&self, state: &mut H) {
+        self.id.hash(state)
+    }
+}
+
+impl<Message> PartialEq for Mailbox<Message> {
+    fn eq(&self, other: &Self) -> bool {
+        self.id.eq(&other.id)
+    }
+}
+
+impl<Message> Eq for Mailbox<Message> {}
+
+impl<Message> Mailbox<Message> {
+    pub fn actor_instance_name(&self) -> String {
+        format!("{}:{}", self.actor_name, self.id)
+    }
+
+    pub(crate) async fn send_actor_message(
+        &self,
+        msg: ActorMessage<Message>,
+    ) -> Result<(), SendError> {
+        self.sender.send_async(msg).await?;
+        Ok(())
+    }
+
+    /// Send a message to the actor synchronously.
+    ///
+    /// SendError is returned if the user is already terminated.
+    ///
+    /// (See also [Self::send_blocking()])
+    pub(crate) async fn send_message(&self, msg: Message) -> Result<(), SendError> {
+        self.send_actor_message(ActorMessage::Message(msg)).await
+    }
+
+    /// Send a message to the actor in a blocking fashion.
+    /// When possible, prefer using [Self::send()].
+    pub(crate) fn send_message_blocking(&self, msg: Message) -> Result<(), SendError> {
+        self.sender.send(ActorMessage::Message(msg))?;
+        Ok(())
+    }
+
+    pub fn send_command_blocking(&self, command: Command<Message>) -> Result<(), SendError> {
+        self.command_sender.send(command)?;
+        Ok(())
+    }
+
+    pub async fn send_command(&self, command: Command<Message>) -> Result<(), SendError> {
+        self.command_sender.send_async(command).await?;
+        Ok(())
+    }
+}
+
+pub struct Inbox<Message> {
+    pub rx: flume::Receiver<ActorMessage<Message>>,
+    command_rx: flume::Receiver<Command<Message>>,
+}
+
+#[derive(Debug)]
+pub enum ReceptionResult<M> {
```

Comment:
Can we rename this enum `ReceiveResult` or `RecvResult`? `ReceptionResult` could be both interpreted as the result of trying to send or receive something. Looks like we're only using this enum for the latter here, so we might as well make it clearer.

Reply:
It was entirely remove in a refactoring that really simplified the mailbox.